test_check_dependencies failures on 1.3.0

```python
================================================================================================================= FAILURES =================================================================================================================
______________________________________________________________________________________________________ test_check_dependencies[False] ______________________________________________________________________________________________________

mocker = <pytest_mock.plugin.MockerFixture object at 0x7f86d59586e0>, package_test_flit = '/build/python-build/src/python-build/tests/packages/test-flit'

    def test_check_dependencies(mocker, package_test_flit):
        mocker.patch('pyproject_hooks.BuildBackendHookCaller.get_requires_for_build_sdist')
        mocker.patch('pyproject_hooks.BuildBackendHookCaller.get_requires_for_build_wheel')

        builder = build.ProjectBuilder(package_test_flit)

        side_effects = [
            [],
            ['something'],
            pyproject_hooks.BackendUnavailable,
        ]

        builder._hook.get_requires_for_build_sdist.side_effect = copy.copy(side_effects)
        builder._hook.get_requires_for_build_wheel.side_effect = copy.copy(side_effects)

        # requires = []
>       assert builder.check_dependencies('sdist') == {('flit_core<4,>=2',)}
E       AssertionError: assert set() == {('flit_core<4,>=2',)}
E
E         Extra items in the right set:
E         ('flit_core<4,>=2',)
E
E         Full diff:
E         + set()
E         - {
E         -     (
E         -         'flit_core<4,>=2',
E         -     ),
E         - }

tests/test_projectbuilder.py:267: AssertionError
------------------------------------------------------------------------------------------------------------ Captured log call -------------------------------------------------------------------------------------------------------------
INFO     build:_builder.py:221 Getting build dependencies for sdist...
========================================================================================================= short test summary info ==========================================================================================================
FAILED tests/test_projectbuilder.py::test_check_dependencies[False] - AssertionError: assert set() == {('flit_core<4,>=2',)}

  Extra items in the right set:
  ('flit_core<4,>=2',)

  Full diff:
  + set()
  - {
  -     (
  -         'flit_core<4,>=2',
  -     ),
  - }
============================================================================================================ 1 failed in 0.06s =============================================================================================================
```

Encountered when trying to build the new package for Arch Linux.

Deps:
```text
core/mpdecimal                4.0.1-1 
core/python                   3.13.5-1
extra/python-packaging        25.0-1  
extra/python-pyproject-hooks  1.2.0-3 

extra/perl-error                   0.17030-2 
extra/perl-mailtools               2.22-2    
extra/perl-timedate                2.33-8    
extra/python-autocommand           2.2.2-7   
extra/python-distlib               0.3.9-2   
extra/python-iniconfig             2.1.0-1   
extra/python-jaraco.collections    5.1.0-1   
extra/python-jaraco.context        6.0.1-1   
extra/python-jaraco.functools      4.1.0-1   
extra/python-jaraco.text           4.0.0-2   
extra/python-more-itertools        10.7.0-1  
extra/python-platformdirs          4.3.6-2   
extra/python-pluggy                1.6.0-1   
extra/python-pygments              2.19.2-1  
extra/uv                           0.8.4-1   
extra/zlib-ng                      2.2.4-1   
extra/git                          2.50.1-3  
extra/python-build                 1.2.2-3   
extra/python-filelock              3.18.0-1  
extra/python-flit-core             3.12.0-1  
extra/python-installer             0.7.0-10  
extra/python-pytest                1:8.4.1-1 
extra/python-pytest-mock           3.14.0-2  
extra/python-pytest-rerunfailures  15.1-1    
extra/python-setuptools            1:80.9.0-1
extra/python-setuptools-scm        8.3.1-1   
extra/python-uv                    0.8.4-1   
extra/python-virtualenv            20.28.0-1 
extra/python-wheel                 0.45.1-1  
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -92,6 +92,7 @@ typing = [
   { include-group = "extra" },
 ]
 dev = [
+  "flit-core",
   { include-group = "test" },
   { include-group = "typing" },
 ]

--- a/tests/test_projectbuilder.py
+++ b/tests/test_projectbuilder.py
@@ -8,7 +8,7 @@ import pathlib
 import sys
 import textwrap
 
-from typing import Never
+from typing import NoReturn
 
 import pyproject_hooks
 import pytest
@@ -250,7 +250,7 @@ def test_build_missing_backend(packages_path, distribution, tmpdir):
         builder.build(distribution, str(tmpdir))
 
 
-def _nothing_installed(name: str) -> Never:
+def _nothing_installed(name: str) -> NoReturn:
     raise _importlib.metadata.PackageNotFoundError(name)
 
 

--- a/tests/test_projectbuilder.py
+++ b/tests/test_projectbuilder.py
@@ -8,6 +8,8 @@ import pathlib
 import sys
 import textwrap
 
+from typing import Never
+
 import pyproject_hooks
 import pytest
 
@@ -248,9 +250,15 @@ def test_build_missing_backend(packages_path, distribution, tmpdir):
         builder.build(distribution, str(tmpdir))
 
 
-def test_check_dependencies(mocker, package_test_flit):
+def _nothing_installed(name: str) -> Never:
+    raise _importlib.metadata.PackageNotFoundError(name)
+
+
+def test_check_dependencies(mocker, package_test_flit, monkeypatch):
     mocker.patch('pyproject_hooks.BuildBackendHookCaller.get_requires_for_build_sdist')
     mocker.patch('pyproject_hooks.BuildBackendHookCaller.get_requires_for_build_wheel')
+
+    monkeypatch.setattr(_importlib.metadata, 'distribution', _nothing_installed)
 
     builder = build.ProjectBuilder(package_test_flit)
 

